crypto/Kconfig

```diff
@@ -17,14 +17,15 @@ config NRF_CC310_BL
 	bool "nrf_cc310_bl HW crypto library for nRF devices with CryptoCell CC310."
 	select NRFXLIB_CRYPTO
 	select FPU # The cc310_bl lib uses floating point registers.
-	depends on !NORDIC_SECURITY_BACKEND
+	depends on IS_SECURE_BOOTLOADER || MCUBOOT
```

Comment:
It's bad practice for subsystems to know of their users.

But doing this properly would require a rewrite that is beyond the scope of code review.